### PR TITLE
Some unit tests for MSBuild telemetry

### PR DESCRIPTION
**Pull request by jeffkl**
Also added a commit that I didn't push with the original PR which has some more try...catches.

The `MSBuildForwardingApp` is hard to test because it has a private `ForwardingApp` which has private arguments.  So I used reflection in order to verify the functionality.  The only other option I saw was to make stuff `internal` instead of `private` or expose some properties to just tests; both of which I wasn't sure if I was allowed to do.

### CODE REVIEW

**Comment by livarcocc** [comment]
@dotnet-bot Test CentOS7.1 x64 Debug Build please
@dotnet-bot Test Windows_NT x86 Debug Build please

File: src/dotnet/commands/dotnet-msbuild/MSBuildLogger.cs [hunk 1]
```diff
@@ -17,36 +16,58 @@ public sealed class MSBuildLogger : Logger
 
         public MSBuildLogger()
         {
-            string sessionId = Environment.GetEnvironmentVariable(MSBuildForwardingApp.TelemetrySessionIdEnvironmentVariableName);
+            try
+            {
+                string sessionId = Environment.GetEnvironmentVariable(MSBuildForwardingApp.TelemetrySessionIdEnvironmentVariableName);
```
Comment (livarcocc) [inline src/dotnet/commands/dotnet-msbuild/MSBuildLogger.cs]: NIT: break the line here. Keep it at less than 120 characters. Horizontal scrolling is very inconvenient.

File: test/dotnet-msbuild.Tests/GivenDotnetMSBuildBuildsProjects.cs [hunk 2]
```diff
@@ -69,5 +77,60 @@ public void When_help_is_invoked_Then_MSBuild_extra_options_text_is_included_in_
...
         }
 
+        [Fact]
+        public void WhenTelemetryIsEnabledTheLoggerIsAddedToTheCommandLine()
+        {
+            string[] allArgs = GetArgsForMSBuild(() => true);
+
+            allArgs.Should().NotBeNull();
+
+            allArgs.Should().Contain(value => value.StartsWith("\"/Logger:"), "The MSBuild logger argument should be specified when telemetry is enabled.");
```
Comment (livarcocc) [inline test/dotnet-msbuild.Tests/GivenDotnetMSBuildBuildsProjects.cs]: NIT: line lenght.

Comment (livarcocc) [inline test/dotnet-msbuild.Tests/GivenDotnetMSBuildBuildsProjects.cs]: Check it out every where in your code changes please.

**Comment by livarcocc** [comment]
@dotnet-bot Test CentOS7.1 x64 Debug Build

**Comment by piotrpMSFT** [comment]
Also fixes https://github.com/dotnet/cli/issues/4588

**Pull request merged**